src/edu/ucsb/nceas/metacat/dataone/quota/BookKeeperClient.java

```diff
@@ -0,0 +1,471 @@
+/**
+ *  Copyright: 2020 Regents of the University of California and the
+ *             National Center for Ecological Analysis and Synthesis
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+package edu.ucsb.nceas.metacat.dataone.quota;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.Date;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.message.BasicHeader;
+import org.apache.wicket.util.io.IOUtils;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.dataone.bookkeeper.api.Quota;
+import org.dataone.bookkeeper.api.QuotaList;
+import org.dataone.bookkeeper.api.Usage;
+import org.dataone.bookkeeper.api.UsageList;
+import org.dataone.configuration.Settings;
+import org.dataone.service.exceptions.InvalidRequest;
+import org.dataone.service.exceptions.NotFound;
+import org.dataone.service.exceptions.ServiceFailure;
+import org.dataone.service.types.v1.Subject;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+/**
+ * A client class to access the BookKeeper service
+ * @author tao
+ *
+ */
+public class BookKeeperClient {
+    public static final int DEFAULT_REMOTE_USAGE_ID = -1;
+    
+    private static final String QUOTAS = "quotas";
+    private static final String QUOTATYPE = "quotaType";
+    private static final String QUOTAID = "quotaId";
+    private static final String QUOTASUBJECT = "subject";
+    private static final String REQUESTOR = "requestor";
+    private static final String USAGES = "usages";
+    private static final String INSTNACEID = "instanceId";
```

Comment:
Change references to this too:
```suggestion
    private static final String INSTANCEID = "instanceId";
```

Reply:
See the commit 5cd4fedd911965286c05975ede3ae941f1695e49.